### PR TITLE
Add wasm modules

### PR DESCRIPTION
**Pull request by dsherret**

### CODE REVIEW

File: reference_gen/web-categories.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
   "Streams": "Manage data streams, queuing strategies, and transformations. Handle data in chunks, process large datasets, and optimize memory usage.\n\nEg {@linkcode ReadableStream}, {@linkcode WritableStream}, {@linkcode TransformStream}",
   "Temporal": "Date and time handling. Includes long-lived Workflows, calendar systems, time zones, and precise duration calculations.\n\nEg {@linkcode Temporal.Now}, {@linkcode Temporal.PlainDate}",
   "URL": "Manipulate URLs, extract data from URLs and manage query parameters.\n\nEg {@linkcode URL}",
-  "WASM": "Efficiently execute computationally intensive tasks. WASM module compilation, instantiation, memory management, and interaction with imports and exports.\n\nEg {@linkcode WebAssembly.instantiate}, {@linkcode WebAssembly.Module}, {@linkcode WebAssembly.Instance}",
+  "Wasm": "Efficiently execute computationally intensive tasks. Wasm module compilation, instantiation, memory management, and interaction with imports and exports.\n\nEg {@linkcode WebAssembly.instantiate}, {@linkcode WebAssembly.Module}, {@linkcode WebAssembly.Instance}",
```
Comment (dsherret) [inline reference_gen/web-categories.json:19]: https://github.com/denoland/deno/pull/26954/files

File: runtime/reference/wasm.md [hunk 2]
```diff
@@ -9,15 +9,174 @@ oldUrl:
...
-[browsers provide](https://developer.mozilla.org/en-US/docs/WebAssembly).
+[browsers provide](https://developer.mozilla.org/en-US/docs/WebAssembly) and by
+importing them as modules.
 
-## Using WebAssembly in Deno
+## Wasm modules
 
-To run WebAssembly in Deno, all you need is a binary to run. The following
-binary exports a `main` function that just returns `42` upon invocation:
+Starting in Deno 2.1, WebAssembly modules can be imported and its use type
```
Comment (bartlomieju) [inline runtime/reference/wasm.md:20]:
```suggestion
Starting in Deno 2.1, WebAssembly modules can be imported and their use type
```

**Review by bartlomieju — APPROVED** [review]

**Comment by dsherret** [comment]
Will wait on the release before merging in order to pick up https://github.com/denoland/deno/pull/26954/files

**Comment by bartlomieju** [comment]
@dsherret you can now merge it

**Pull request merged**